primitives/core/src/crypto.rs

```diff
@@ -457,6 +458,11 @@ impl<T: Sized + AsMut<[u8]> + AsRef<[u8]> + Public + Derive> Ss58Codec for T {
 	}
 }
 
+// Use the default implementations of the trait in serde feature.
+// The std implementation is not available because of std only crate Regex.
+#[cfg(all(not(feature = "std"), feature = "serde"))]
+impl<T: Sized + AsMut<[u8]> + AsRef<[u8]> + Public + Derive> Ss58Codec for T {}
```

Comment:
From what I see this implementation is not correct. It cannot deserialize derived keys correctly.

Not sure what to do here: we could try to parse w/o regex, look for no-std regex. Probably not implementing `serde` for `Derive` is not an option as we need it in no_std. @bkchr what would be your advice here?

Reply:
To my understanding, (de)serialization only uses `to_ss58check`, no string functions involved. I don't think it would compile to `no_std` otherwise, because both (reimplemented) functions `from_string_with_version` and `from_string` of the `Ss58Codec` trait are not available in `std` anyway. (see https://github.com/haerdib/substrate/blob/add-serde-feature-to-primitives/primitives/core/src/crypto.rs#L315-L322 & https://github.com/haerdib/substrate/blob/add-serde-feature-to-primitives/primitives/core/src/crypto.rs#L355-L358)

Reply:
Derive requires specialised from_string implementation:
https://github.com/haerdib/substrate/blob/3b30d526915727143544116ed044d2e5c43a6aaf/primitives/core/src/crypto.rs#L429-L445

The sample use case is here:
https://github.com/haerdib/substrate/blob/3b30d526915727143544116ed044d2e5c43a6aaf/primitives/core/src/crypto.rs#L1390

I think maybe we can accept this limitation for now, and open follow-up issue to implement parsing for nostd.

Reply:
Another thought, I think it makes more sense: let's make `serde` implementation `from_string` `unimplemented!` and open a new issue to fix that in future.

Probably it will be better to be explicit about missing functionality, then providing some partial/not-working solution.

Reply:
Yes we can use `unimplemented!` for now. However, you will probably require this for the `GenesisConfig` work.

Reply:
I have added the unimplemented macro in https://github.com/paritytech/substrate/pull/13027/commits/0e00dd62ea86813adb6eeff90581575abcb24f47. But please note that it will now generate a runtime error instead of a compile error. Not sure if that's user friendlier.

Reply:
Thank you. I prefer explicit failure over potentially not failing buggy behavior.
Ticket for improvement: https://github.com/paritytech/substrate/issues/14081